client_server_test.go

```diff
@@ -1074,8 +1082,10 @@ func TestDialRemainingHeaderBytes(t *testing.T) {
 				return
 			}
 		})
+	cstDialer := cstDialer
+	cstDialer.MaxHeaderSize = 256
 	ws, _, err := cstDialer.Dial(s.URL, nil)
```

Comment:
The problem here is that we need the message read into the headerBr at the time of reading the header. This is not deterministic. Will be nice to make this test repeat the dial/ReadMessage some 20x to get a good chance of hitting the case under test.